presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -365,6 +372,25 @@ public void enforceMemoryLimits()
     }
 
     /**
+     * Prune extraneous info from old queries
+     */
+    public void pruneExpiredQueries()
+    {
+        if (expirationQueue.size() <= maxQueryHistory) {
+            return;
+        }
+
+        int count = 0;
+        // we're willing to keep full info for up to maxQueryHistory queries
+        for (QueryExecution query : expirationQueue) {
```

Comment:
expirationQueue is oldest-to-newest (i.e., the order in which queries should be expired), so this will prune the info for the `N-maxQueryHistory` newest queries.